Sources/SwiftDiagnostics/GroupedDiagnostics.swift

```diff
@@ -57,6 +57,10 @@ public struct GroupedDiagnostics {
   /// source file IDs.
   var rootIndexes: [Syntax: SourceFileID] = [:]
 
+  /// Diagnostics that aren't associated with any source file because they have
+  /// no location information.
+  var floatingDiagnostics: [Diagnostic] = []
```

Comment:
Instead of introducing a new term _floating_ here, how about calling this something like `unknownPositionDiagnostics`?